modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/Sidebar/Sidebar.es.js

```diff
@@ -385,17 +407,15 @@ class Sidebar extends Component {
 			useShim: false,
 		});
 
-		this._eventHandler.add(
-			this._dragAndDrop.on(Drag.Events.START, this._handleDragStarted),
-			this._dragAndDrop.on(DragDrop.Events.END, this._handleDragEnded),
-			this._dragAndDrop.on(
-				DragDrop.Events.TARGET_ENTER,
-				this._handleDragTargetEnter
-			),
-			this._dragAndDrop.on(
-				DragDrop.Events.TARGET_LEAVE,
-				this._handleDragTargetLeave
-			)
+		this._dragAndDrop.on(Drag.Events.START, this._handleDragStarted);
```

Comment:
Are these the same events as in https://github.com/liferay-frontend/liferay-portal/pull/684/files#diff-00d6f8807d663a0cebcf10c24bd234910b3ef8c20ef42e87b55445fc3582b33eR162? Did you mean to delete this?

Reply:
They're the same event.
In the `created` method I declared them as variables so that I could call `removeListener()` on them.
And then in the `_bindDragAndDrop` method I just removed `this._eventHandler` wrapper around them and kept the events there.

This pattern is sketchy to me, so I wouldn't mind some clarification. The reason I went with this approach is because I *assumed* that the events in `_bindDragAndDrop` aren't accessible from inside the `created` method.

But I was meaning to test that more today and establish a pattern.

Reply:
I'll let you play around with this a bit. Add some logging or debugging and see if you can verify if the events are being properly added and removed.

Keep in mind that `on` is like `addEventListener` but returns an object that you can use to `removeEventListener`. Think about what happens if you do `foo.addEventListener('click', () => {})` twice.

Reply:
Good tip, thanks @jbalsas

Reply:
So I've reached a conclusion that we don't need to explicitly remove listeners in the `disposeInternal` method because `super.disposeInternal();` removes the events from the `this._dragAndDrop` object. And that's how it's done in [`withMoveableFields.es.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/FormBuilder/withMoveableFields.es.js#L29)

Reply:
> So I've reached a conclusion that we don't need to explicitly remove listeners in the disposeInternal method because super.disposeInternal(); removes the events from the this._dragAndDrop

Could you walk me through the reasoning? You might be right, but you're not proving you are :)

Reply:
I've logged `this._dragAndDrop` before and after `super.disposeInternal()`and looked at `this._dragAndDrop.events`.
Before disposing, it contained all the events, after it was `null`. I took that + the fact that it was a preexisting pattern as gospel and ran with it.

Reply:
> I've logged this._dragAndDrop before and after super.disposeInternal()and looked at this._dragAndDrop.events.
Before disposing, it contained all the events, after it was null.

Next time, please add that information so anyone can see what you saw ;)

> I took that + the fact that it was a preexisting pattern as gospel and ran with it

This is not proof. Debugging and getting one sample of state is not proof that the code is doing what you say is doing. It's an indicator you might be right, but you should try to understand what's happening and why. Otherwise, you risk getting a lot of false positives because "I clicked here and there and this was null" might not hold true in other scenarios.

Please, take a look at `disposeInternal` in `Drag` and follow the trail to see if you can verify if your assertion is correct and why.

Reply:
Okay, I'll dig deeper here, I agree with your sentiment!

Reply:
I went down the rabbit hole and reached the `EventEmitter`'s `disposeInternal` and it's definition of `events` [here](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-events/src/EventEmitter.js#L19-L24):
```js
/**
 * Holds event listeners scoped by event type.
 * @type {Object<string, !Array<!function()>>}
 * @protected
 */
this.events_ = null; // this is where it's initially defined
```

```js
/**
 * Disposes of this instance's object references.
 * @override
 */
disposeInternal() {
	this.events_ = null;
}
```

### After testing
I've tested further the difference between manually removing listeners and relying on the `disposeInternal()`. Here's the report:
- Manually removing event listeners goes to `this._dragAndDrop.events_.eventName` and sets its value to `null`
- `disposeInternal()` doesn't go that extra step and just nullifies the entire `this._dragAndDrop.events_` object
- No other property of the `this._dragAndDrop` object contains our events, so I believe that relying on `disposeInternal()` is enough because it deletes everything related to events, and the events are attached to the `this._dragAndDrop` object

P.S. This is the only method I know of testing if events exists in a case like this, so if I missed a specific way of testing for events, I didn't know that way exists.

Reply:
So, you're confident now that if `dratAndDrop` is `disposed` then the event listeners that were attached to it using `on` -> `addSingleListener` are also dropped? Do you feel more confident now about your stance with this PR?

Reply:
I am confident, yes. As the only place those events are mentioned in the object are inside `events_` which get nullified when `dragAndDrop` is disposed.

Reply:
I'm marking this as resolved because it apparently is, and it's hard to read the code otherwise.